### PR TITLE
error handling and cleanup

### PR DESCRIPTION
**Pull request by glaslos**
Go lint cleanup and some improved error handling

### CODE REVIEW

**Comment by glaslos** [comment]
bump?

**Comment by glaslos** [comment]
bump?

File: encoder.go [hunk 1]
```diff
@@ -2,9 +2,9 @@ package midi
 
 import (
 	"encoding/binary"
-	"errors"
 	"io"
-	"log"
+
+	"github.com/pkg/errors"
```
Comment (mattetti) [inline encoder.go:7]: adding this dependency what was preventing me from merging this PR (that I honestly forgot about). Any chance we can avoid adding an external dependency?

File: decoder_event.go [hunk 2]
```diff
@@ -421,12 +421,16 @@ func (p *Decoder) parseMetaMsg(e *Event) (nextChunkType, bool, error) {
 			}
 
 			// key
+<<<<<<< HEAD
```
Comment (mattetti) [inline decoder_event.go:424]: oops :)

Comment (mattetti) [reply decoder_event.go:424]: thanks @glaslos for reviving that old PR I totally forgot about. It looks like there was a little merge problem.

**Comment by glaslos** [comment]
Should be good now :)

**Comment by mattetti** [comment]
thank you!

**Pull request merged**